modules/apps/fragment/fragment-web/src/main/java/com/liferay/fragment/web/internal/portlet/action/UploadFragmentCompositionPreviewMVCActionCommand.java

```diff
@@ -36,20 +45,67 @@ protected void doProcessAction(
 			ActionRequest actionRequest, ActionResponse actionResponse)
 		throws Exception {
 
-		_uploadHandler.upload(
-			_fragmentEntryPreviewImageEditorUploadFileEntryHandler,
-			_itemSelectorUploadResponseHandler, actionRequest, actionResponse);
+		uploadHandler.upload(
+			fragmentEntryPreviewImageEditorUploadFileEntryHandler,
+			itemSelectorUploadResponseHandler, actionRequest, actionResponse);
 	}
 
+	protected FragmentEntryPreviewImageEditorUploadFileEntryHandler
+		fragmentEntryPreviewImageEditorUploadFileEntryHandler =
+			new FragmentEntryPreviewImageEditorUploadFileEntryHandler();
+
+	@Reference
+	protected ItemSelectorUploadResponseHandler
+		itemSelectorUploadResponseHandler;
+
 	@Reference
-	private FragmentEntryPreviewImageEditorUploadFileEntryHandler
-		_fragmentEntryPreviewImageEditorUploadFileEntryHandler;
+	protected UploadHandler uploadHandler;
 
 	@Reference
-	private ItemSelectorUploadResponseHandler
-		_itemSelectorUploadResponseHandler;
+	private DLAppService _dlAppService;
+
+	@Reference(
+		target = "(resource.name=" + FragmentConstants.RESOURCE_NAME + ")"
+	)
+	private PortletResourcePermission _portletResourcePermission;
```

Comment:
@regiCesar21 this can not be right, UploadFragmentCompositionPreviewMVCActionCommand is the base class of UploadFragmentEntryPreviewMVCActionCommand now, it can not have private field with ```@Reference ``` annotation. This will cause some SCR scanning warn/error.

CC @rafaprax